test/tests/functional/pbs_qsub_remove_files.py

```diff
@@ -198,7 +198,7 @@ def test_qalter_remove_files(self):
         alter the job with -Roe and check whether it is
         reflecting in qstat -f output.
         """
-        mydate = int(time.time()) + 60
+        mydate = int(time.time()) + 120
```

Comment:
we are not doing much between submitting the job and altering the attribute. 60 seconds seems enough. Are we sure that the test is failing because the job started running.

Reply:
@prakashcv13: This issue was observed during one of the regressions. The test failed while verifying the job's state
_**Error:**_
2018-09-10 **15:39:55**,611 INFO job: executable set to /bin/sleep with arguments: 100
2018-09-10 15:39:55,612 INFOCLI x40-64-centos64: sudo -H -u pbsuser /opt/pbs/bin/qsub -a 09101540 – /bin/sleep 100
2018-09-10 15:39:59,592 INFO submit to x40-64-centos64 as pbsuser: job 1.x40-64-centos64 OrderedDict([('Execution_Time', '**09101540**')])
2018-09-10 15:40:01,033 INFO alter on x40-64-centos64: 1.x40-64-centos64 {'Remove_Files': 'oe'}
2018-09-10 15:40:01,033 INFOCLI x40-64-centos64: /opt/pbs/bin/qalter -R oe 1.x40-64-centos64
2018-09-10 15:40:01,075 INFOCLI x40-64-centos64: /opt/pbs/bin/qstat -f 1.x40-64-centos64
2018-09-10 15:40:01,146 INFO expect on server x40-64-centos64: job_state = W job 1.x40-64-centos64 got: job_state = Q
2018-09-10 15:40:01,649 INFOCLI x40-64-centos64: /opt/pbs/bin/qmgr -c set server scheduling=True
2018-09-10 15:40:01,703 INFOCLI x40-64-centos64: /opt/pbs/bin/qstat -f 1.x40-64-centos64
2018-09-10 **15:40:01**,831 INFO expect on server x40-64-centos64: job_state = W job 1.x40-64-centos64 attempt: 2 got: job_state = R
2018-09-10 15:40:02,332 INFOCLI x40-64-centos64: /opt/pbs/bin/qmgr -c set server scheduling=True

- In the test we are capturing the current time,rounding off the value(using int) and in the test we are adding up 1 min to the above captured time and passing this value to qsub option in -a parameter(which is job's start time)

- If we observe the logs, the current time is "15:39:55", that means the round off value would be "15:39", so accordingly the job's start time should be 15:40 which is 5 secs after the current time

Reply:
time.time() returns no.of seconds since epoch as a float. (int) will round it off to the nearest int value - it does not return a string formatted to "hh:mm:ss" - see example below -
>>> time.time()
**1563977785.850314**
>>> int(time.time())
**1563977786**                       //Here it rounded it to a "higher" value
>>> time.time()
**1563977789.368164**
>>> int(time.time())
**1563977789**                       //Here it truncated to a "lower" value
>>> 
In the logs that you have shown, it is clear to me that time.time() returned something that was equivalent to 15:39:<some seconds>.
adding 60 to it would make it 15:40:<some seconds>

But the format that is being passed to strftime() on line 205 is '%m%d%H%M'. It ignores the seconds part completely, so we see only 15:40.
If we use '%m%d%H%M.%S', we will have 15:40:<some seconds>, the job will remain in 'W' state and that will fix the issue at hand.

However, the test case just needs the job to not be running, which can be simply achieved by disabling the scheduling of jobs, it is not about testing the '-a' option of qsub, so we can go ahead with @dinesh1306's suggestion.

Reply:
Thanks @prakashcv13 for clarification. I have changed the test accordingly. Kindly review